cache/cache_data.go

```diff
@@ -116,13 +116,13 @@ func (c *cache) Set(key []byte, val []byte) bool {
 
 // Get value
 func (c *cache) Get(key []byte) []byte {
-	c.lock.Lock()
+	c.lock.RLock()
```

Comment:
This is incorrect, because we update LRU below with listUnlink/listAppend and this is a write operation.

Reply:
Yeah I've noticed that just when Travis failed.
Give me a sec to implement a solution which targets both

Reply:
00e2bceb0837b7e0510f9eb9bd3ffc9464cbea73 targets this

Was more modification than I spotted initially, sorry for my first quick shot